tests/test_formats.py

```diff
@@ -128,6 +129,30 @@ def test_defaults_with_path(self):
             with tsinfer.load(filename) as other:
                 self.assertEqual(other, input_file)
 
+    def test_bad_max_file_size(self):
+        with tempfile.TemporaryDirectory(prefix="tsinf_format_test") as tempdir:
+            filename = os.path.join(tempdir, "samples.tmp")
+            for bad_size in ['a', '', -1, [1, 3], np.array([1, 2])]:
+                self.assertRaises(
+                    (ValueError, TypeError), formats.SampleData,
+                    path=filename, max_file_size=bad_size)
+
+    @unittest.skipIf(sys.platform == "win32",
+                     "windows simultaneous file permissions issue")
+    def test_too_small_max_file_size(self):
+        with tempfile.TemporaryDirectory(prefix="tsinf_format_test") as tempdir:
+            base_size = 2**16  # Big enough to allow the initial file to be created
+            filename = os.path.join(tempdir, "samples.tmp")
+            # Fail immediately if the max_size is so small we can't even create a file
+            self.assertRaises(
+                 lmdb.MapFullError, formats.SampleData,
+                 path=filename, sequence_length=1, max_file_size=1)
+            sample_file = formats.SampleData(
+                    path=filename, sequence_length=1, max_file_size=base_size)
+            self.assertRaises(
+                lmdb.MapFullError, sample_file.add_site, 0,
```

Comment:
Do we always flush the data to file straightaway here? Might be better to do:
```python
with self.assertRaises(lmdb.MapFullError):
    sample_file.add_site, 0, alleles=['0', '1'], genotypes=np.zeros(base_size, dtype=np.int8))
    sample_file.flush() # Or whatever we call it.
```

Reply:
Oh, I've never used the `with self.assertRaises()` format in anger before. Thanks.